src/plugins/intel_cpu/src/graph.cpp

```diff
@@ -766,77 +714,108 @@ void Graph::AllocateWithReuse(const std::vector<size_t>& syncNodesInds) {
-            box.size = div_up(boxSize, alignment);
-            definedBoxes.push_back(box);
-        } else {
-            box.size = boxSize;
-            undefinedBoxes.push_back(box);
-        }
+        memoryRegions.push_back(reg);
     }
 
-    // Process defined boxes (static shapes)
-    ov::MemorySolver staticMemSolver(definedBoxes);
-    size_t total_size = static_cast<size_t>(staticMemSolver.solve()) * alignment;
-
-    memWorkspace = std::make_shared<Memory>(getEngine(), DnnlBlockedMemoryDesc(ov::element::i8, Shape(VectorDims{total_size})));
+    // special processing of the dynamic output edges
+    auto it = std::remove_if(memoryRegions.begin(), memoryRegions.end(), [&](const MemoryRegion& region) {
+        if (region.size >= 0 || !one_of(region.type, MemoryRegion::RegionType::OUTPUT, MemoryRegion::RegionType::IO)) {
+            return false;
+        }
+        bool result = false;
+        for (auto& edge : edge_clusters[region.id]) {
+            auto child = edge->getChild();
+            if (child->getType() == Type::Output && edge->getStatus() == Edge::Status::NeedAllocation) {
+                auto proxyMemBlock = std::make_shared<ProxyMemoryBlock>();
+                DEBUG_LOG("ProxyMemoryBlock ", proxyMemBlock, " ", this);
+                edge->allocate(proxyMemBlock);
+
+                // Store the output memory blocks.
+                // So that, the infer requests can be able to access them.
+                int count = 0;
+                for (auto& output : outputNodesMap) {
+                    if (output.second == child) {
+                        outputNodesMemBlocksMap[output.first] = proxyMemBlock;
+                        count++;
+                    }
+                }
+                // sometimes there are unused output ports.
+                OPENVINO_ASSERT(count <= 1, "CPU plugin cannot find output node. count ", count);
+                result = true;
+            }
+        }
+        return result;
+    });
 
-    if (edge_clusters.empty())
-        return;
+    memoryRegions.erase(it, memoryRegions.end());
 
-    auto* workspace_ptr = static_cast<int8_t*>(memWorkspace->getData());
+    //Set up the memory control subsystem.
+    this->m_pMemoryControl = &(getGraphContext()->getNetworkMemoryControl()->createMemoryControlUnit(syncNodesInds));
+    auto memoryBlocks = m_pMemoryControl->insert(memoryRegions);
 
-    for (const auto& box : definedBoxes) {
+    // attach all the not yet allocated edges to the memory contol
+    for (auto&& item : memoryBlocks) {
         int count = 0;
-        for (auto& edge : edge_clusters[box.id]) {
+        for (auto&& edge : edge_clusters[item.first]) {
             if (edge->getStatus() == Edge::Status::NeedAllocation) {
-                int64_t offset = staticMemSolver.get_offset(box.id);
-                // !! Fallback to individual memory allocation !!
-                // if you like to check infer without reuse just call this function without arguments.
```

Comment:
Just wondering if the same debug functionality is still available

Reply:
Honestly it isn't, since to have such option we need to replace the memory manager used for the dynamic tensors allocation. However, such change is rather trivial.